Mathlib/Algebra/Ring/Associator.lean

```diff
@@ -0,0 +1,91 @@
+/-
+Copyright (c) 2025 Bernhard Reinke. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Bernhard Reinke
+-/
+import Mathlib.Algebra.Ring.Defs
+import Mathlib.Algebra.Group.Hom.End
+import Mathlib.Algebra.Group.Hom.Instances
+import Mathlib.Tactic.Abel
+
+/-!
+## Associator in a ring
+If `R` is a non-associative ring, then  `(x * y) * z - x * (y * z)` is called the `associator` of
+ring elements `x y z : R`. We realize the associator as a map that is an `AddMonoidHom` in each
+argument. The associator vanishes exactly when `R` is associative. We prove a similar statement for
+semirings by directly comparing the maps `mulLeft₃` and `mulRight₃`, the `AddMonoidHom` versions of
+the multiplications `(x * y) * z` and `x * (y * z)`.
+-/
+
+universe u
+
+namespace AddMonoidHom
+
+section NonUnitalNonAssocSemiring
+variable {R : Type u} [NonUnitalNonAssocSemiring R]
+
+/-- The multiplication `(x * y) * z` of three elements of a (non-associative)
+(semi)-ring is an `AddMonoidHom` in each argument. -/
+def mulLeft₃ : R →+ R →+ R →+ R where
+  toFun x := comp mul (mulLeft x)
+  map_zero' := by ext; simp
+  map_add' := fun x y => by ext; simp [add_mul]
+
+@[simp]
+theorem mulLeft₃_apply (x y z : R) : mulLeft₃ x y z = (x * y) * z := rfl
+
+/-- The multiplication `x * (y * z)` of three elements of a (non-associative)
+(semi)-ring is an `AddMonoidHom` in each argument. -/
+def mulRight₃ : R →+ R →+ R →+ R where
+  toFun x := compr₂ mul (mulLeft x)
+  map_zero' := by ext; simp
+  map_add' := fun x y => by ext; simp [add_mul]
+
+@[simp]
+theorem mulRight₃_apply (x y z : R) : mulRight₃ x y z = x * (y * z) := rfl
+
+/-- An a priori non-associative semiring is associative if the `AddMonoidHom` versions of
+the multiplications `(x * y) * z` and `x * (y * z)` agree. -/
+theorem associative_iff_mulLeft₃_eq_mulRight₃ : Std.Associative (fun (x y : R) ↦ x * y) ↔
+  mulLeft₃ (R := R) = mulRight₃ := by
+  constructor
```

Comment:
Could you turn this iff around? Currently the RHS is more complicated than the LHS